File: api/src/main/kotlin/com/example/demohackaton/entity/Shop.kt
package com.example.demohackaton.entity

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*

@Entity
@Table(name = "shop")
data class Shop(
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  val id: Long,
  val name: String,
  val description: String? = null,
  val img: String? = null,
  @JsonManagedReference
  @OneToMany(mappedBy = "shop", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
  val products: List<Product> = ArrayList(),
)

File: frontend/src/types/shop.ts
export type TProduct = {
  id: number;
  name: string;
  description: string;
  img: string;
  price: number;
  stock: number;
};

export type TShop = {
  id: number;
  name: string;
  description: string;
  img: string;
  products: TProduct[];
};
